cornflow-server/cornflow/shared/authentication/auth.py

```diff
@@ -118,73 +120,63 @@ def generate_token(user_id: int = None) -> str:
-    ) -> dict:
-        """
-        This method takes a token issued by an OID provider, the relevant information about the OID provider
-        and validates that the token was generated by such source, is valid and extracts the information
-        in the token for its use during the login process
-
-        :param str token: the received token
-        :param str client_id: the identifier from the client
-        :param str tenant_id: the identifier for the tenant
-        :param str issuer: the identifier for the issuer of the token
-        :param int provider: the identifier for the provider of the token
-        :return: the decoded token as a dictionary
-        :rtype: dict
-        """
-        public_key = self._get_public_key(token, tenant_id, provider)
         try:
-            decoded = jwt.decode(
-                token,
-                public_key,
-                verify=True,
-                algorithms=["RS256"],
-                audience=[client_id],
-                issuer=issuer,
-            )
-            return decoded
+            # First try to decode header to validate basic token structure
+            unverified_payload = jwt.decode(token, options={"verify_signature": False})
+            issuer = unverified_payload.get("iss")
+            
+            # For internal tokens
+            if issuer == INTERNAL_TOKEN_ISSUER:
+                payload = jwt.decode(
+                    token, current_app.config["SECRET_TOKEN_KEY"], algorithms="HS256"
+                )
+                return {"username": payload["sub"]}
+
         except jwt.ExpiredSignatureError:
-            raise InvalidCredentials(
-                "The token has expired, please login again",
-                log_txt="Error while trying to validate a token. The token has expired.",
-            )
-        except jwt.InvalidTokenError:
-            raise InvalidCredentials(
-                "Invalid token, please try again with a new token",
-                log_txt="Error while trying to validate a token. The token is not valid.",
+                raise InvalidCredentials(
+                    "The token has expired, please login again",
+                    log_txt="Error while trying to decode token. The token has expired.",
+                    status_code=400
+                )
+        except jwt.InvalidTokenError as e:
+                raise InvalidCredentials(
+                    "Invalid token format or signature",
+                    log_txt="Error while trying to decode token. The token format is invalid.",
+                    status_code=400
+                )
+
+                
+        # For OpenID tokens
+        if current_app.config["AUTH_TYPE"] == AUTH_OID:
+            decoded = Auth().verify_token(
+                token, 
+                current_app.config["PROVIDER_URL"], 
+                current_app.config["EXPECTED_AUDIENCE"]
             )
+            return {"username": decoded["sub"]}
```

Comment:
For me this structure is a little bit weird. Either the try except only covers the `jwt.decode` call or we cover also the other decode call (the one I guess is inside `verify_token`) so that we can have the same error handling